SocketStats/main.cpp

```diff
@@ -0,0 +1,152 @@
+            }
+
+            if (NSAPI_TCP == stats[i].proto) {
+                printf("%-15s", "TCP");
+            } else {
+                printf("%-15s", "UDP");
+            }
+            printf("%-15d", stats[i].sent_bytes);
+            printf("%-15d", stats[i].recv_bytes);
+            printf("%-15lld\n", stats[i].last_change_tick);
+        }
+        num++;
+        ThisThread::sleep_for(SAMPLE_TIME);
+    }
+}
+
+#include "mbed.h"
+
+// Network interface
+NetworkInterface *net;
+
+int main()
+{
+    // Request the shared queue
+    Thread *thread = new Thread(osPriorityNormal1, 2048);
+    int remaining;
+    int rcount;
+    char *p;
+    char *buffer = new char[256];
+    nsapi_size_or_error_t result;
+
+    thread->start(print_stats);
+    // Bring up the ethernet interface
+    printf("Mbed OS Socket example\n");
+
+#ifdef MBED_MAJOR_VERSION
+    printf("Mbed OS version: %d.%d.%d\n\n", MBED_MAJOR_VERSION, MBED_MINOR_VERSION, MBED_PATCH_VERSION);
+#endif
+
+    net = NetworkInterface::get_default_instance();
+
+    if (!net) {
+        printf("Error! No network inteface found.\n");
+        return 0;
+    }
+
+    result = net->connect();
+    if (result != 0) {
+        printf("Error! net->connect() returned: %d\n", result);
+        return result;
+    }
+
+    // Show the network address
+    const char *ip = net->get_ip_address();
+    const char *netmask = net->get_netmask();
+    const char *gateway = net->get_gateway();
+    printf("IP address: %s\n", ip ? ip : "None");
+    printf("Netmask: %s\n", netmask ? netmask : "None");
+    printf("Gateway: %s\n", gateway ? gateway : "None");
+
+    // Open a socket on the network interface, and create a TCP connection to api.ipify.org
+    TCPSocket socket;
+    // Send a simple http request
+    char sbuffer[] = "GET / HTTP/1.1\r\nHost: api.ipify.org\r\nConnection: close\r\n\r\n";
+    nsapi_size_t size = strlen(sbuffer);
+
+    result = socket.open(net);
+    if (result != 0) {
+        printf("Error! socket.open() returned: %d\n", result);
+    }
+
+    result = socket.connect("api.ipify.org", 80);
+    if (result != 0) {
+        printf("Error! socket.connect() returned: %d\n", result);
+        goto DISCONNECT;
+    }
+
+    // Loop until whole request sent
+    while(size) {
+        result = socket.send(sbuffer+result, size);
+        if (result < 0) {
+            printf("Error! socket.send() returned: %d\n", result);
+            goto DISCONNECT;
+        }
+        size -= result;
+        printf("sent %d [%.*s]\n", result, strstr(sbuffer, "\r\n")-sbuffer, sbuffer);
+    }
+
+    // Receieve an HTTP response and print out the response line
+    remaining = 256;
+    rcount = 0;
+    p = buffer;
+    while (0 < (result = socket.recv(p, remaining))) {
+        p += result;
+        rcount += result;
+        remaining -= result;
+    }
+    if (result < 0) {
+        printf("Error! socket.recv() returned: %d\n", result);
+        goto DISCONNECT;
```

Comment:
I avoid them 👍 but this is copy paste code of https://github.com/ARMmbed/mbed-os-example-sockets. Just integrated stats into available example to demonstrate how can we use them :-)

Reply:
That it is! If you'd like to throw up a PR to other repo to change this and the other comment above that'd be cool but up to you. Not going to stop this PR in the meantime as it is just using the existing Socket example as is.

Question: Do we have a mechanism in place to ensure the socket example and this one stay in sync? And does it particularly matter if they do get out of sync as long as it's functional?

Reply:
> That it is! If you'd like to throw up a PR to other repo to change this and the other comment above that'd be cool but up to you

Sure.. will add it to my list.. :-)

>  Do we have a mechanism in place to ensure the socket example and this one stay in sync?

I don't have automated mechanism in place now, but this can go in list as well for future... Sorry

Reply:
> Do we have a mechanism in place to ensure the socket example and this one stay in sync?

Nope. If anything, this repo is more of a holding bucket until we figure out the right place for these examples.